statistics/handle/update.go

```diff
@@ -181,7 +184,10 @@ func (s *SessionStatsCollector) Delete() {
 func (s *SessionStatsCollector) Update(id int64, delta int64, count int64, colSize *map[int64]int64) {
 	s.Lock()
 	defer s.Unlock()
-	s.mapper.update(id, delta, count, colSize)
+	// if table locked, skip
+	if !isTableLocked(s.tableLocked, id) {
+		s.mapper.update(id, delta, count, colSize)
+	}
```

Comment:
We don't record the change of `count` and `modify_count` when the table is locked. After we unlock the table, even if the real `modify_count` / `count` is high, we don't trigger auto analyze since we lose the change of `count` and `modify_count`. Is it the expected behavior?

Reply:
After discussing with @pingandb and @winoros, maybe we can record the `count` and `modify_count` at the lock time into `mysql.stats_table_locked` and read `count` from `mysql.stats_table_locked` during the lock period. Another way is to update `modify_count` and `count` delta to `mysql.stats_table_locked` rather than `mysql.stats_meta`.

Reply:
yeah, we should store the stats delta during table locked, and I save the delta value to mysql.stats_table_locked.  The way that reading count and modify_count from mysql.stats_meta with lock time in mysql.stats_table_locked maybe a bit more complicated, since there are many places to read from mysql.stats_meta.